### PR TITLE
Set preload=false if not using overlay2 as docker storage driver

### PR DESCRIPTION
**Pull request by priyawadhwa**
fixes https://github.com/kubernetes/minikube/issues/7626#event-3577309612

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -1158,3 +1159,26 @@ func getKubernetesVersion(old *config.ClusterConfig) string {
...
+// validateDockerStorageDriver checks that docker is using overlay2
+// if not, set preload=false (see #7626)
+func validateDockerStorageDriver(drvName string) {
+	if !driver.IsKIC(drvName) {
+		return
+	}
+	if _, err := exec.LookPath("docker"); err != nil {
+		exit.WithCodeT(exit.BadUsage, "Please make sure docker is available on your PATH")
+	}
+	j, err := exec.Command("docker", "info", "--format", "{{.Driver}}").Output()
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go:1172]: Please use oci.CachedDaemonInfo

Since docker dameon is expensive.

Comment (medyagh) [reply cmd/minikube/cmd/start.go:1172]: as part of this you would need to add Driver to 
```
type SysInfo struct {
	CPUs        int    // CPUs is Number of CPUs
	TotalMemory int64  // TotalMemory Total available ram
	OSType      string // container's OsType (windows or linux)
}

```
https://github.com/medyagh/minikube/blob/0701f5ed35142c77964160c2d8999c20b0982c8c/pkg/drivers/kic/oci/info.go#L29

Comment (medyagh) [reply cmd/minikube/cmd/start.go:1172]: $ sudo podman info | grep overla
  GraphDriverName: overlay

Comment (medyagh) [reply cmd/minikube/cmd/start.go:1172]: for Podman the field is in

```
            "version": "runc version 1.0.0~rc6+dfsg1\ncommit: 1.0.0~rc6+dfsg1-3\nspec: 1.0.1"
        },
        "SwapFree": 0,
        "SwapTotal": 0,
        "arch": "amd64",
        "cpus": 2,
        "eventlogger": "journald",
        "hostname": "podman-exp-temp",
        "kernel": "4.19.0-8-cloud-amd64",
        "os": "linux",
        "rootless": false,
        "uptime": "2258h 28m 11.68s (Approximately 94.08 days)"
    },
    "registries": {
        "search": [
            "docker.io",
            "quay.io"
        ]
    },
    "store": {
        "ConfigFile": "/etc/containers/storage.conf",
        "ContainerStore": {
            "number": 1
        },
        "GraphDriverName": "overlay",
        "GraphOptions": {},
        "GraphRoot": "/var/lib/containers/storage",
        "GraphStatus": {
            "Backing Filesystem": "extfs",
            "Native Overlay Diff": "true",
            "Supports d_type": "true",
            "Using metacopy": "false"
        },
        "ImageStore": {
            "number": 2
        },
        "RunRoot": "/var/run/containers/storage",
        "VolumePath": "/var/lib/containers/storage/volumes"
    }
}

```

Comment (priyawadhwa) [reply cmd/minikube/cmd/start.go]: thanks medya! switched to using DaemonInfo

**Pull request merged**